### PR TITLE
Feat/list functionality

### PR DESCRIPTION
**Pull request by nattsnb**

### CODE REVIEW

File: LinkedList.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  remove(elementNumber) {
+    const elementToRemove = this.getNthElement(elementNumber);
+    elementToRemove.nextMember.previousMember = elementToRemove.previousMember;
+    elementToRemove.previousMember.nextMember = elementToRemove.nextMember;
+  }
+
+  reverse = () => {
+    if (this.#tail && this.#head !== this.#tail) {
```
Comment (Solnick) [inline LinkedList.js:43]: This function should mutate the LinkedList itself not return a new LinekedList instance

File: LinkedList.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        previousElement,
+      );
+      newMember.value = value;
+      previousElement.nextMember = newMember;
+      nextElement.previousMember = newMember;
+    }
+  }
+  getNthElement(elementNumber) {
+    let element = this.#head;
+    for (let i = 0; i < elementNumber - 1; i++) {
```
Comment (Solnick) [inline LinkedList.js]: What if there are 3 elements and you provide 5 as an argument?

File: LinkedList.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      let reversedList = new LinkedList();
+      let currentMember = this.#tail;
+      while (currentMember !== null) {
+        reversedList.addToTail(currentMember.value);
+        currentMember = currentMember.previousMember;
+      }
+      return reversedList;
+    }
+  };
+  switchPositions(leftElementNumber, rightElementNumber) {
```
Comment (Solnick) [inline LinkedList.js]: as arguments pass a reference to the LinkedListElement, this way you don't need to call the `getNthElement` which is consuming N operations.

File: LinkedList.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        const rightElementPreviousMemberSaved = rightElement.previousMember;
+        const leftElementNextMemberSaved = leftElement.nextMember;
+        rightElement.previousMember = leftElement.previousMember;
+        leftElement.nextMember = rightElement.nextMember;
+        leftElement.previousMember = rightElementPreviousMemberSaved;
+        rightElement.nextMember = leftElementNextMemberSaved;
+      }
+    }
+  }
+  addBetween(previousElementNumber, nextElementNumber, value) {
```
Comment (Solnick) [inline LinkedList.js]: pass references to the LinkedListElements as arguments instead of the number of the element

**Review by Solnick — CHANGES_REQUESTED** [review]

File: LinkedList.js (hunk 3 above)
Comment (nattsnb) [reply LinkedList.js]: done

File: LinkedList.js (hunk 4 above)
Comment (nattsnb) [reply LinkedList.js]: done

File: LinkedList.js (hunk 2 above)
Comment (nattsnb) [reply LinkedList.js]: done

File: LinkedList.js (hunk 1 above)
Comment (nattsnb) [reply LinkedList.js:43]: done

**Pull request merged**